### DISCUSSION TITLE
Image.Load seems to be very slow, how can I increase it's performance?

### DISCUSSION THREAD

**Discussion by SalahAkbari** [SixLabors/ImageSharp#1474] [Q&A] [answered]
I need to load some images, about 70, from database and then choose a random item from the list and combine it with another image which I have already generated:

```
Images = urlPathsEntity.Select(c => Image.Load(c.Url)).ToList() // 70 Images

int i = 0;

while (i < 11000)
{
    var Image = Images[rand.Next(Images.Count)];
    Image.Mutate(x => x.Resize(x, y));
    oldImage.Mutate(o => o.DrawImage(Image, new Point(0, col), 1f));

    while (j < 11000)
    {
         Image = Images[rand.Next(Images.Count)];
         Image.Mutate(x => x.Resize(x, y));
         oldImage.Mutate(o => o.DrawImage(Image, new Point(j, col), 1f));

         j += 1300;
    }

    col += 390;

    i += 390;
    j = 1300;
}

await oldImage.SaveAsync(path);
```

Well, actually with this approach, the performance is not acceptable and generating the final image takes about 15-20 seconds, I thought it will be a bad idea to load all of the 70 images from database and url path and then select a random from them, so I changed my code like this:

```
ImagesPath = urlPathsEntity.ToList() // 70 Images Path

int i = 0;

while (i < 11000)
{
    var imagePath = ImagesPath[rand.Next(ImagesPath.Count)];
    using var Image = await Image.LoadAsync(imagePath.Url);

    Image.Mutate(x => x.Resize(x, y));
    oldImage.Mutate(o => o.DrawImage(Image, new Point(0, col), 1f));

    while (j < 11000)
    {
         var imagePath2 = ImagesPath[rand.Next(ImagesPath.Count)];
         using var Image2 = await Image.LoadAsync(imagePath2.Url);

         Image2 = Images[rand.Next(Images.Count)];
         Image2.Mutate(x => x.Resize(x, y));
         oldImage.Mutate(o => o.DrawImage(Image2, new Point(j, col), 1f));

         j += 1300;
    }

    col += 390;

    i += 390;
    j = 1300;
}

```

Well, this time I didn't load up all images and I am just loading one image from the random path, but since it is inside a while loop, the total performance is not good still, I would say, it is even worse than the old one. I was really confused, is there any way I can improve my overall performance?

**Comment by antonfirsov** [comment, marked as answer]
@SalahAkbari you are doing approximately `(11000/390) * (11000 / 1300) == 28 * 8 == 224` decodes in your second code instead of 70, so not surprising it's slower.

Loading 70 images will be relatively slow with any framework/library. What is the format of your images? How large they are typically?

**Reply by JimBobSquarePants** [reply]
I can't see any disposal code either.

**Reply by SalahAkbari** [reply]
@JimBobSquarePants I have used the `using` statements in my second code. I thought, wrapping the class in a using(...) {} will make sure any unmanaged resources are got rid of as necessary.
